src/inc/dnmd.h

```diff
@@ -1,10 +1,37 @@
 #ifndef _SRC_INC_DNMD_H_
 #define _SRC_INC_DNMD_H_
 
+#include <stdint.h>
+#include <stddef.h>
+
+#ifndef __cplusplus
+
+#include <stdbool.h>
```

Comment:
I would move this up to the other standard headers (that is, outside of the C++ guard). All implementations should be smart about guarding including this header.